wp-cli/class-theme-check-cli.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+if ( defined( 'WP_CLI' ) && WP_CLI ) {
+	$parent_dir = dirname( __DIR__ );
+	require_once $parent_dir . '/checkbase.php';
+	require_once $parent_dir . '/main.php';
+	WP_CLI::add_command( 'theme-check', 'Theme_Check_Command' );
+}
+
+class Theme_Check_Command extends WP_CLI_Command {
+	/**
+	 * Run a theme check on the specified theme or the current theme.
+	 *
+	 * ## OPTIONS
+	 * [<theme>]
+	 * : The slug of the theme to check. If not provided, checks the current theme.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: cli
+	 * options:
+	 *   - cli
+	 *   - json
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *     # Check the current active theme
+	 *     wp theme-check run
+	 *
+	 *     # Check a specific theme
+	 *     wp theme-check run twentytwentyfour
+	 *
+	 *     # Check the current active theme and output results as JSON
+	 *     wp theme-check run --format=json
+	 *
+	 *     # Check a specific theme and output results as JSON
+	 *     wp theme-check run twentytwentyfour --format=json
+	 *
+	 * @param array $args       Indexed array of positional arguments.
+	 * @param array $assoc_args Associative array of options.
+	 * @return void
+	 */
+	public function run( $args, $assoc_args ) {
+		$format = \WP_CLI\Utils\get_flag_value( $assoc_args, 'format', 'cli' );
+
+		// Get the current theme
+		$current_theme      = wp_get_theme();
+		$current_theme_slug = $current_theme->get_stylesheet();
+
+		// Use the provided theme slug if available, otherwise use the current theme
+		$check_theme_slug = ! empty( $args[0] ) ? $args[0] : $current_theme_slug;
+
+		// Get the theme
+		$theme = wp_get_theme( $check_theme_slug );
+
+		if ( ! $theme->exists() ) {
+			if ( $format === 'json' ) {
+				$json_output = array(
+					'completed' => false,
+					'result'    => "Error: Theme '{$check_theme_slug}' not found.",
+					'messages'  => array(),
+				);
+				WP_CLI::line( wp_json_encode( $json_output, JSON_PRETTY_PRINT ) );
+				return;
+			}
+			WP_CLI::error( "Theme '{$check_theme_slug}' not found." );
+		}
+
+		// Run the checks.
+		$success = run_themechecks_against_theme( $theme, $check_theme_slug );
+
+		if ( $format === 'json' ) {
+			if ( ! $success ) {
+				$json_output = array(
+					'completed' => false,
+					'result'    => "Error: Theme check failed for {$check_theme_slug}.",
+					'messages'  => array(),
+				);
+				WP_CLI::line( wp_json_encode( $json_output, JSON_PRETTY_PRINT ) );
+				return;
+			}
+			$this->display_themechecks_as_json( $check_theme_slug );
+		} else {
+			if ( ! $success ) {
+				WP_CLI::error( "Theme check failed for {$check_theme_slug}." );
+			}
+			$this->display_themechecks_in_cli( $check_theme_slug );
```

Comment:
Execution stops if a `WP_CLI::error` is thrown, so the errors are not being displayed. Could this line be moved above the $success check?

Reply:
I'm not sure, i'm seeing this logic here:
```php
// If the check process failed (if the checks were not able to run properly) raise an error. There's nothing else to do.
if ( ! $success ) {
WP_CLI::error( "Theme check failed for {$check_theme_slug}." );
}
// If the checking process run print in the console the checks. This will also print the things marked as an error in the theme.
$this->display_themechecks_in_cli( $check_theme_slug );
```
Could you explain why you think we should move it?

Reply:
```
// If the check process failed (if the checks were not able to run properly) raise an error. There's nothing else to do.
```

I don't think this is correct. `$success` is the result of `run_themechecks_against_theme` which returns `false` if the theme fails a check. 

> Could you explain why you think we should move it?

Using `twentytwentyfive`, here is the output currently: 

```
Error: Theme check failed for twentytwentyfive.
```

Moving call to `display_themechecks_in_cli` above the conditional, the result is:

```
Success: Theme check completed for twentytwentyfive.

REQUIRED: phpcs.xml.dist XML file found. This file must not be in the production version of the theme.

WARNING: assets/images/image-from-rawpixel-id-2222755.webp is 997.6 KB in size. Large file sizes have a negative impact on website performance and loading time. Compress images before using them.

WARNING: assets/images/image-from-rawpixel-id-2224378.webp is 879.6 KB in size. Large file sizes have a negative impact on website performance and loading time. Compress images before using them.

WARNING: assets/images/location.webp is 778.1 KB in size. Large file sizes have a negative impact on website performance and loading time. Compress images before using them.

WARNING: assets/images/poster-image-background.webp is 574.7 KB in size. Large file sizes have a negative impact on website performance and loading time. Compress images before using them.

INFO Themes that use the tag accessibility-ready will need to undergo an accessibility review. See https://make.wordpress.org/themes/handbook/review/accessibility/

WARNING: Your theme appears to be in the wrong directory for the theme name. The directory name must match the slug of the theme. This theme's correct slug and text-domain is twenty-twenty-five. (If this is a child theme, you can ignore this error.)

INFO: Only one text-domain is being used in this theme. Make sure it matches the theme's slug correctly so that the theme will be compatible with WordPress.org language packs. The domain found is twentytwentyfive.

RECOMMENDED: Requires PHP is recommended to have major and minor versions only (e.g. 7.4). Patch version is not needed (e.g. 7.4.1).

Error: Theme check failed for twentytwentyfive.
```

I think the latter is more useful because it reports what checks failed (the first one marked `REQUIRED`). 

Also I am not sure we should throw an error here if `$success` isn't true, because the theme check successfully ran, but the result was it failed the checks.

Reply:
This command's focus should be on pass/fail check for a theme, rather than run is successfully completed or not. And exit code should be returned based on that I believe.

Reply:
I think I could implement your feedback, please see: https://github.com/WordPress/theme-check/pull/458#issuecomment-2349033538